### PR TITLE
Us 12 us 13 change personal details

### PR DESCRIPTION
**Pull request by dawidk-kainos**

### CODE REVIEW

File: src/main/java/domain/Persons.java [hunk 1]
```diff
@@ -48,11 +53,20 @@
...
+    @Column(name = "default_start_hour")
+    private Time defaultStartHour;
+
+    @Column(name = "default_finish_hour")
+    private Time defaultFinishHour;
+
+    @JsonIgnore
     private Boolean active;
 
+    @JsonIgnore
```
Comment (scyganska) [inline src/main/java/domain/Persons.java]: Front-end will need this serialized.

**Comment by RogowskaMalgorzata** [comment]
Seems ok.

File: src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            <label class="col-lg-3 control-label">Tester:</label>
+            <div class="col-lg-1">
+                <input type="checkbox" ng-model="details.isTester">
+            </div>
+        </div>
+        <div class="form-group">
+            <label class="col-lg-3 control-label">Band level:</label>
+            <div class="col-lg-2">
+                <div class="ui-select">
+                    <select id="user_time_zone" class="form-control">
```
Comment (jraniszewski) [inline src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html]: Why there is 'user_time_zone' id in Band level ?

File: src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                        <option value="1">1</option>
```
Comment (jraniszewski) [inline src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html]: We should help interviewers understand what those numbers mean.

File: src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            <label class="col-lg-3 control-label">Default starting hour:</label>
+            <div class="col-lg-2">
+                <input type="time" name="input" ng-model="details.startHour"
+                       placeholder="HH:mm:ss" min="00:00:00" max="24:00:00" required />
+            </div>
+        </div>
+        <div class="form-group">
+            <label class="col-lg-3 control-label">Default finishing hour:</label>
+            <div class="col-lg-2">
+                <input type="time" name="input" ng-model="details.finishHour"
```
Comment (jraniszewski) [inline src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html]: Name should be finish hour.

File: src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                        <option value="6">6</option>
+                    </select>
+                </div>
+            </div>
+        </div>
+
+        <div class="form-group">
+            <label class="col-lg-3 control-label">Default starting hour:</label>
+            <div class="col-lg-2">
+                <input type="time" name="input" ng-model="details.startHour"
```
Comment (jraniszewski) [inline src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html]: Name should be start hour.

File: src/main/java/resources/PersonResources.java [hunk 6]
```diff
@@ -7,6 +7,8 @@
 import domain.Slots;
```
Comment (wojciechh-kainos) [inline src/main/java/resources/PersonResources.java:9]: This class should be named PersonsResource

File: src/main/resources/assets/html/partials/interviewer/tdp-interviewer-home.html [hunk 7]
```diff
@@ -47,4 +44,5 @@ <h4 class="navigate">{{startDate}} - {{endDate}}</h4>
     <div>
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/html/partials/interviewer/tdp-interviewer-home.html]: Do these buttons need to be inside their own divs?

File: src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            <div class="col-lg-1">
+                <input type="checkbox" ng-model="person.isTest">
+            </div>
+        </div>
+        <div class="form-group">
+            <label class="col-lg-4 control-label">Band level:</label>
+            <div class="col-lg-2">
+                <div class="ui-select">
+                    <select class="form-control" ng-model="person.bandLevel">
+                        <option value="1">1 - Principal</option>
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html]: These options should have values assigned from an enum

File: src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+                        <option value="6">6 - Trainee</option>
+                    </select>
+                </div>
+            </div>
+        </div>
+        <div class="form-group">
+            <label class="col-lg-4 control-label">Default starting hour:</label>
+            <div class="col-lg-2">
+                <input type="time" name="defaultStartHour" ng-model="person.defaultStartHour"
+                       placeholder="HH:mm:ss" min="00:00:00" max="24:00:00" required />
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html]: min should probably be 08:00 and max 10:00

File: src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+            </div>
+        </div>
+        <div class="form-group">
+            <label class="col-lg-4 control-label">Default finishing hour:</label>
+            <div class="col-lg-2">
+                <input type="time" name="defaultFinishHour" ng-model="person.defaultFinishHour"
+                       placeholder="HH:mm:ss" min="00:00:00" max="24:00:00" required />
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html]: min should be 16:00, max 18:00

File: src/main/resources/assets/js/application/interviewer/controllers/tdprInterviewerDetailsController.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
+define(['angular', 'application/interviewer/tdprInterviewerModule'], function (angular, tdprInterviewerModule) {
+    tdprInterviewerModule.controller("tdprInterviewerDetailsController", function ($scope, $stateParams, tdprPersonService, $filter, $state, Notification) {
+
+        function init() {
+            tdprPersonService.getPersonDetails($stateParams.id).then(function (response) {
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/interviewer/controllers/tdprInterviewerDetailsController.js]: If we're retrieving person details in every single controller, maybe we should move this to resolve?

File: src/main/resources/assets/js/application/interviewer/controllers/tdprInterviewerHomeController.js [hunk 12]
```diff
@@ -68,12 +68,8 @@ define(['angular', 'application/interviewer/tdprInterviewerModule', 'application
...
             var startDate = $filter('date')(getDayOfTheWeek(new Date(), $scope.relativeDayNumber), "dd-MM-yyyy");
             var endDate = $filter('date')(getDayOfTheWeek(new Date(), 4 + $scope.relativeDayNumber), "dd-MM-yyyy");
-            tdprSlotsService.updateSlots(slots, id, startDate, endDate).then(function () {
-                $scope.showSubmitSuccess = true;
-                $timeout(function () {
-                    $scope.showSubmitSuccess = false;
-                }, 2000);
-            });
+            tdprSlotsService.updateSlots(slots, id, startDate, endDate).then(function(){
+                Notification.success({message: 'Changes saved!', delay: 2000});});
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/interviewer/controllers/tdprInterviewerHomeController.js]: This is a good refactor 👍

File: src/main/resources/assets/js/application/interviewer/controllers/tdprInterviewerHomeController.js [hunk 13]
```diff
@@ -93,6 +89,10 @@ define(['angular', 'application/interviewer/tdprInterviewerModule', 'application
         $scope.markSlots = function(slot) {
             slot.type === $scope.currentType ? slot.type = AvailabilityEnum.empty.id : slot.type = $scope.currentType
         };
+        $scope.goDetails = function(){
+              $state.go('tdpr.interviewer.details', {'id' : id});
+        }
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/interviewer/controllers/tdprInterviewerHomeController.js:212]: ; missing at the end

File: src/main/resources/assets/js/application/interviewer/services/tdprPersonService.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
+define(['angular', 'application/interviewer/tdprInterviewerModule'], function(angular, tdprInterviewerModule) {
+    tdprInterviewerModule.service("tdprPersonService", function($http) {
+
+        this.getPersonDetails = function(id) {
+            return $http.get("/api/person/" + id);
+        };
+
+        this.updatePersonDetails = function(person){
+            return $http.put("/api/person/create", person);
+        }
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/interviewer/services/tdprPersonService.js]: ; missing at the end

File: src/main/resources/assets/js/application/interviewer/tdprInterviewerModuleConfig.js [hunk 15]
```diff
@@ -16,7 +18,15 @@ define(['angular' 
                         controller: "tdprInterviewerHomeController" 
                     } 
                 }
-            }); 
+            }).state("tdpr.interviewer.details", {
+              url: "/details/{id}",
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/interviewer/tdprInterviewerModuleConfig.js]: url should be  /{id}/details

File: src/main/resources/migrations.xml [hunk 16]
```diff
@@ -12,6 +12,7 @@
     <include file="changelogs/initialChangeColumnDayNameInSlots.xml"/>
     <include file="changelogs/initialPopulatePersonsTable.xml"/>
     <include file="changelogs/initialPopulateSlotsTable.xml"/>
+    <include file="changelogs/initialAddDefaultWorkingHoursPersonsTable.xml"/>
```
Comment (wojciechh-kainos) [inline src/main/resources/migrations.xml]: Why was this file added here and not at the end?

File: src/main/resources/assets/js/application/interviewer/services/tdprPersonService.js [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+
+        this.getPersonDetails = function(id) {
+            return $http.get("/api/person/" + id).then(function(response){
+                person = response.data;
+                return response;
+            });
+        };
+
+        this.updatePersonDetails = function(person){
+            return $http.put("/api/person/create", person);
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/interviewer/services/tdprPersonService.js]: a PUT to an endpoint called create? This is peculiar and not compliant with REST rules. a put should be to /api/person/{id} if you want to update

File: src/main/resources/assets/js/application/interviewer/tdprInterviewerModuleConfig.js [hunk 18]
```diff
@@ -16,7 +18,21 @@ define(['angular'
...
+                  "@": {
+                      templateUrl: "/html/partials/interviewer/tdpr-interviewer-details.html",
+                      controller: "tdprInterviewerDetailsController"
+                  }
+              },
+              resolve:{
+                  service: "tdprPersonService",
+                  promise: function(service, $stateParams){
+                      return service.getPersonDetails($stateParams.id);
+                  }
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/interviewer/tdprInterviewerModuleConfig.js:32]: we are resolving a person - it should look like this:
resolve: {
  person: function(tdprPersonService, $stateParams) {
    return: tdprPersonService.getPersonDetails($stateParams.personId);
  }
}

File: src/main/resources/changelogs/initialAddDefaultWorkingHoursPersonsTable.xml [hunk 19]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+         http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd">
+
+    <changeSet id="12" author="dawidkre">
```
Comment (wojciechh-kainos) [inline src/main/resources/changelogs/initialAddDefaultWorkingHoursPersonsTable.xml]: I've seen this id used in another pull request - can we clarify?

File: src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+                <input type="time" name="defaultFinishHour" ng-model="person.defaultFinishHour"
+                       placeholder="HH:mm:ss" min="16:00:00" max="18:00:00" required />
+                <span ng-show="detailsForm.defaultFinishHour.$error.time">Invalid. </span>
+                <span ng-show="detailsForm.defaultFinishHour.$error.min || detailsForm.defaultFinishHour.$error.max">Values between 16:00 and 18:00!</span>
+            </div>
+        </div>
+        <div class="form-group">
+            <label class="col-lg-2 control-label"></label>
+            <div class="col-lg-6">
+                <input type="button" class="btn btn-primary" value="Save Changes" ng-click="updateDetails()">
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html]: class should always be the first attribute

File: src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html [hunk 21]
```diff
@@ -0,0 +1,57 @@
+<br>
+<div class="well">
+    <form name="detailsForm" class="form-horizontal col-lg-offset-3" role="form">
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/html/partials/interviewer/tdpr-interviewer-details.html]: class should always be the first attribute

File: src/main/resources/assets/js/application/interviewer/controllers/tdprInterviewerDetailsController.js [hunk 22]
```diff
@@ -0,0 +1,37 @@
+define(['angular', 'application/interviewer/tdprInterviewerModule'], function (angular, tdprInterviewerModule) {
+    tdprInterviewerModule.controller("tdprInterviewerDetailsController", function ($scope, $stateParams, tdprPersonService, $filter, $state, Notification, BandLevelEnum) {
+
+        $scope.BandLevelEnum = BandLevelEnum;
+
+        function init() {
+                $scope.person = tdprPersonService.getPerson();
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/interviewer/controllers/tdprInterviewerDetailsController.js]: Why are you using tdprPersonService here instead of injecting person - that's why we're using resolve

File: src/main/resources/app_config.yml [hunk 23]
```diff
@@ -12,13 +12,13 @@ database:
   driverClass: org.postgresql.Driver
 
   # the username
-  user: tdp
+  user: postgres
```
Comment (wojciechh-kainos) [inline src/main/resources/app_config.yml]: This probably shouldn't be committed

File: src/main/resources/app_config.yml [hunk 24]
```diff
@@ -47,4 +47,4 @@ smtpConfig:
   port: 25
   from: no-reply@kainos.com
 
-domain: http://localhost:8888
\ No newline at end of file
+domain: http://localhost:8888/#/interviewer/home/
```
Comment (wojciechh-kainos) [inline src/main/resources/app_config.yml]: This probably shouldn't be changed
